synapse/api/errors.py

```diff
@@ -82,6 +93,11 @@ class Codes:
     UNREDACTED_CONTENT_DELETED = "FI.MAU.MSC2815_UNREDACTED_CONTENT_DELETED"
 
 
+# `Codes` used to be a namespace for codes. This is now replaced
+# with the enum `Code` but we maintain it for backwards compatibility.
+Codes = Code
```

Comment:
I still don't see why this is necessary. I don't really see a value in renaming `Codes`, and it sounds to me like it's going to create maintenance burden by creating inconsistencies throughout the codebase (most other enums in Synapse have their names in the plural form).

Can we just stick with `Codes` please?

Reply:
My bad, I hadn't noticed these plural enums. I had never encountered such beasts :)

Reply:
Yeah I get that's a bit confusing from an external pov :sweat: